### PR TITLE
Validation & model spec

### PR DESCRIPTION
**Pull request by Said-laasri**
**Add the following validations:**

1. For the User model:
```
- Name must not be blank.
- PostsCounter must be an integer greater than or equal to zero.
```
2. For the Post model:
```
- Title must not be blank.
- Title must not exceed 250 characters.
- CommentsCounter must be an integer greater than or equal to zero.
- LikesCounter must be an integer greater than or equal to zero.
```
**Add unit specs for all of your models' methods and validations.**

### CODE REVIEW

File: db/migrate/20221020114402_add_user_ref_to_user.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+class AddUserRefToUser < ActiveRecord::Migration[7.0]
+  def change
+    add_reference :likes, :author, null: false, foreign_key: { to_table: :users }
+    add_reference :comments, :author, null: false, foreign_key: { to_table: :users }
+  end
+end
```
Comment (GzimAsani) [inline db/migrate/20221020114402_add_user_ref_to_user.rb:6]: - [x] Kindly check this error that you are getting from this migration
image

**Review by GzimAsani — CHANGES_REQUESTED** [review]
### Hello    @Said-laasri        :wave:   
 Good job so far! There are some issues that you still need to work on to go to the next project but you are almost there!
 
 ## Highlights
 * Good job adding tests

 ## Required Changes 

- [x] Please add tests as as being asked on the requirements `.to_not be_valid` for all of them it is not appropriate.
- [x] Check the comment below :arrow_double_down: 






 ## Optional suggestions
 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding! clapclapclap
 
 Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.
 
 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: db/migrate/20221020114402_add_user_ref_to_user.rb (hunk 1 above)
Comment (Said-laasri) [reply db/migrate/20221020114402_add_user_ref_to_user.rb:6]: @GzimAsani  Thank you for reviewing my project, I will make the necessary corrections as suggested
I deleted the duplicated reference in likes and comments

**Review by ShadyShawkat — APPROVED** [review]
## Hi @Said-laasri, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight
- All linters are passing ✔️ 
- All tests are passing ✔️ 
- Correct use of GitFlow ✔️ 
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 

Image

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**